doc/modules/classes.rst

```diff
@@ -1646,3 +1646,27 @@ Recently deprecated
 
 To be removed in 1.0 (renaming of 0.25)
 ---------------------------------------
+
+.. _loss_function_ref:
+
+:mod:`sklearn._loss`: Private Loss Function Classes
```

Comment:
Is there a specific reason to document these?

I'm not sure we should: we never expose private stuff in the API ref so far, and even though it's clear from here that these are private, it won't be as clear once users land on the individual pages of such losses. Until we have a system to visually distinguish the private stuff from the public stuff in the docs, I would recommend not to document them publicly. (related is our developer-API discussion)

Reply:
I agree.

Reply:
Maybe we could move this to a dedicated page, somewhere under the developer documentation sections of the documentation.

Reply:
I transfered this discussion to the original PR, see https://github.com/scikit-learn/scikit-learn/pull/20567#discussion_r693877622.